pulsar-broker/src/test/java/org/apache/pulsar/compaction/CompactionTest.java

```diff
@@ -1384,7 +1384,7 @@ public void testEmptyPayloadDeletesWhenEncrypted() throws Exception {
 
                 Message<byte[]> message4 = consumer.receive();
                 Assert.assertEquals(message4.getKey(), "key2");
-                Assert.assertEquals(new String(message4.getData()), "");
```

Comment:
@codelipenghui, Could you help take a look here? Is this a break change?

Reply:
I don't think this is a break change. It's a BUG fix. 

Users send a message without a value should be treated as null value, not empty "String".